### DISCUSSION TITLE
BLACK PILL (STM32DUINO) WITH ADAFRUIT ST7789 LIBRARY

### DISCUSSION THREAD

**Discussion by ht93** [stm32duino/Arduino_Core_STM32#1426] [Q&A]
I am attempting to use black pill (weact v3.1, newest stm32core installed through arduino IDE, https://github.com/WeActTC/MiniSTM32F4x1) with 2 inch ST7789 IPS display (https://www.waveshare.com/2inch-lcd-module.htm) using the Adafruit_ST7789 (https://github.com/adafruit/Adafruit-ST ... t_ST7789.h) under Adafruit_GFX. I have attached my code here. My code works fine with the arduino nano. However it does not work on black pill. Arduino IDE successfully uploaded the program to the black pill. But the display is always black (black light is on, but nothing shows). I tried to solve it on my own, like change the pin number and make sure the pin is correctly connected, look into the library but does not know what is wrong. May I know anyone know what wrong with this? Thanks a lot!
```c++
#include <SPI.h>
#include <Adafruit_GFX.h>    // Core graphics library
#include <Adafruit_ST7789.h> // Hardware-specific library for ST7789

#define TFT_CS    PA4
#define TFT_DC    PB0
#define TFT_RST   PB1
#define TFT_MOSI    PA7//PB15
#define TFT_SCLK    PA5//PB13

//SPIClass SPI_1(PA7, PA6, PA5);
//SPIClass SPI_2(PB15, PB14, PB13);

//Adafruit_ST7789 tft = Adafruit_ST7789(TFT_CS, TFT_DC, TFT_RST);
//Adafruit_ST7789 tft = Adafruit_ST7789(&SPI_1, TFT_CS, TFT_DC, TFT_RST);
Adafruit_ST7789 tft = Adafruit_ST7789(TFT_CS, TFT_DC, TFT_MOSI, TFT_SCLK, TFT_RST);

void setup(void) {
  tft.init(240, 320);           // Init ST7789 320x240
  tft.fillScreen(ST77XX_BLACK);
  tft.setRotation(1);
  tft.setTextWrap(false);
  tft.setTextColor(ST77XX_WHITE, ST77XX_BLACK);
  tft.setTextSize(5);
  tft.setCursor(0, 75);
  tft.println("CODE TEST");
  tft.setCursor(0, 125);
  tft.println("HELLO");

  pinMode(PC13, OUTPUT); // LED connect to pin PC13
}

void loop() {
  digitalWrite(PC13, HIGH);   // turn the LED on (HIGH is the voltage level)
  delay(100);               // wait for 100mS
  digitalWrite(PC13, LOW);    // turn the LED off by making the voltage LOW
  delay(100);               // wait for 100mS
}
```
My computer development environment is: Windows 10 19042.1052, arduino IDE 1.8.13, core 2.0.0, upload method DFU, adafruit st7735 and st7789 library 1.7.3, adafruit gfx library 1.10.10

**Comment by fpistm** [comment]
HI @ht93 
I guess this is also your topic:
https://www.stm32duino.com/viewtopic.php?f=7&t=1139
You should avoid to duplicate entries, this will not ease to help you. I advise you to keep only the forum.
I have no clue about this library.

**Reply by ht93** [reply]
I see. thanks for reply. How can I close this?

**Reply by fpistm** [reply]
No need to close. If someone looking for the same topic, he could follow the link to topic on the forum. 😉

**Reply by ht93** [reply]
Great, I added some details about the method I tried with that library in the post. I have tried many methods, even try the circuitpython with blackpill to use adafruit st7789, the display works (so the pin I used in stm32duino and black pill should be correct), but the circuitpy drive (which to store library and code) are extreme small for blackpill (only 31kb in total, and after add display library only 1 kb left).
